hollow/src/main/java/com/netflix/hollow/api/producer/validation/RecordCountChangeValidator.java

```diff
@@ -0,0 +1,125 @@
+        int previousRecordNumber = typeState.getPreviousOrdinals().cardinality();
+
+        float addedPercent = (float) addRecordNumber / previousRecordNumber;
+        float removedPercent = (float) removeRecordNumber / previousRecordNumber;
+        float updatedPercent = (float) updatedRecordNumber / previousRecordNumber;
+
+
+        float addedPercentageThreshold = threshold.addedPercentageThreshold;
+        float removedPercentageThreshold = threshold.removedPercentageThreshold;
+        float updatedPercentageThreshold = threshold.updatedPercentageThreshold;
+
+        ValidationResult.ValidationResultBuilder builder = ValidationResult.from(this);
+        builder.detail("addedRecordNumber", addRecordNumber);
+        builder.detail("removedRecordNumber", removeRecordNumber);
+        builder.detail("updatedRecordNumber", updatedRecordNumber);
+        builder.detail("previousRecordNumber", previousRecordNumber);
+        builder.detail("addedPercentageThreshold", addedPercentageThreshold);
+        builder.detail("removedPercentageThreshold", removedPercentageThreshold);
+        builder.detail("updatedPercentageThreshold", updatedPercentageThreshold);
+
+        boolean pass =
+                (addedPercentageThreshold < 0 || addedPercent < addedPercentageThreshold) &&
+                        (removedPercentageThreshold < 0 || removedPercent < removedPercentageThreshold) &&
+                        (updatedPercentageThreshold < 0 || updatedPercent < updatedPercentageThreshold);
+        if (pass) {
+            return builder.passed();
+        }
+        return builder.failed("record count change is more than threshold");
+    }
+
+    public static class Threshold {
+        private final float removedPercentageThreshold;
+        private final float addedPercentageThreshold;
+        private final float updatedPercentageThreshold;
+
+        Threshold(float removedPercentageThreshold,
+                  float addedPercentageThreshold,
+                  float updatedPercentageThreshold) {
+            this.removedPercentageThreshold = removedPercentageThreshold;
+            this.addedPercentageThreshold = addedPercentageThreshold;
+            this.updatedPercentageThreshold = updatedPercentageThreshold;
+        }
+
+        public static ThresholdBuilder builder() {
+            return new ThresholdBuilder();
+        }
+
+        public static class ThresholdBuilder {
+            private float removedPercentageThreshold = -1f;
+            private float addedPercentageThreshold = -1f;
+            private float updatedPercentageThreshold = -1f;
+
+            public ThresholdBuilder withRemovedPercentageThreshold(float removedPercentageThreshold) {
+                this.removedPercentageThreshold = removedPercentageThreshold;
+                return this;
+            }
+
+            public ThresholdBuilder withAddedPercentageThreshold(float addedPercentageThreshold) {
```

Comment:
since the threshold type is float, some javadoc explaining how the input parameter is interpreted would be helpful here, for e.g. is passing 0.01 treated as 1% or 0.01%? for deleted and updated, we can validate that the input is in the range (0,100]